servers/graphql-kotlin-ktor-server/src/main/kotlin/com/expediagroup/graphql/server/ktor/KtorGraphQLRequestParser.kt

```diff
@@ -61,9 +60,8 @@ class KtorGraphQLRequestParser(
     }
 
     private suspend fun parsePostRequest(request: ApplicationRequest): GraphQLServerRequest? = try {
-        val rawRequest = request.call.receiveText()
-        mapper.readValue(rawRequest, GraphQLServerRequest::class.java)
+        request.call.receive()
```

Comment:
how does the request know to which type the request payload needs to be deserialized to ? is it an inlineClass ?

```kotlin
request.call.receive<GraphQLServerRequest> 
```

Reply:
and yes `receive` is inline reified method